### DISCUSSION TITLE
How to implement on_change debouncing?

### DISCUSSION THREAD

**Discussion by MShekow** [lona-web-org/lona#144] [Q&A] [answered]
Hi, what is the best way to implement debouncing of `CHANGE` events?

**Comment by fscherf** [comment]
I am sorry, what do you mean with "debouncing"?

**Reply by MShekow** [reply]
It is explained well here http://reactivex.io/documentation/operators/debounce.html
The use case is to avoid overloading the backend.

**Comment by fscherf** [comment, marked as answer]
Ah! ``lona.html.TextInput`` does something like that by setting its input timeout. The client then delays all input events and discards them if another event gets fired while the timeout is not met (https://github.com/lona-web-org/lona/blob/master/lona/client/input-events.js#L108)

Currently that's implemented in the client. Do you need that for click events too?

**Reply by MShekow** [reply]
Thank you for elaborating. From the term (`input_delay`) and the documentation it was not obvious that debouncing already happens.

I do not need the functionality for `CLICK` events.

**Reply by fscherf** [reply]
Ah ok! Maybe you can improve the docs on that topic?

**Reply by MShekow** [reply]
See PR https://github.com/lona-web-org/lona/pull/147
